test/integration_test.rb

```diff
@@ -12,6 +12,43 @@ def test_allocations
     assert_valid_firefox_profile(result)
   end
 
+  def test_vernier_run
+    result = vernier_run("--", "ruby", "-e", "'sleep 1'")
+    assert_valid_firefox_profile(result)
+  end
+
+  def test_vernier_run_with_metadata
+    result = vernier_run("--metadata", "key1=val1",
+                         "--metadata", "key2=val2",
+                         "--", "ruby", "-e", "'sleep 1'")
+    assert_valid_firefox_profile(result)
+    assert_equal result["meta"]["userMetadata"]["key1"], "val1"
+    assert_equal result["meta"]["userMetadata"]["key2"], "val2"
+  end
+
+  def test_vernier_run_with_metadata_last_value_takes_precedence
+    result = vernier_run("--metadata", "key1=val1",
+                         "--metadata", "key2=val2",
+                         "--metadata", "key1=val3",
+                         "--", "ruby", "-e", "'sleep 1'")
+    assert_valid_firefox_profile(result)
+    assert_equal result["meta"]["userMetadata"]["key2"], "val2"
+    assert_equal result["meta"]["userMetadata"]["key1"], "val3"
+  end
+
+  # Explicitly call the "vernier run" executable
+  def vernier_run(*argv)
```

Comment:
took inspiration from "run_vernier", with this wrapper that calls "vernier run".